src/main.rs

```diff
@@ -260,15 +262,14 @@ fn run_tracer(child: Pid, config: Config) {
                                     string = format!("{:?}...", truncated_string);
                                 }
                                 output.push_str(string.as_str());
-                                arguments.push(string);
                             }
                             system_call_names::SystemCallArgumentType::Address => {
                                 if value == 0 {
                                     output.push_str("NULL");
-                                    arguments.push(String::from("NULL"));
+                                    arguments.push(serde_json::Value::Null);
                                 } else {
                                     output.push_str(format!("0x{:x}", value as i32).as_str());
-                                    arguments.push(format!("0x{:x}", value as i32));
+                                    arguments.push((value as i32).into());
```

Comment:
To be frank, the cast to `i32` seems incorrect here anyways. In particular this cuts off part of the address in both branches?